erpnext/stock/doctype/stock_ledger_entry/stock_ledger_entry.py

```diff
@@ -181,3 +181,4 @@ def on_doctype_update():
 
 	frappe.db.add_index("Stock Ledger Entry", ["voucher_no", "voucher_type"])
 	frappe.db.add_index("Stock Ledger Entry", ["batch_no", "item_code", "warehouse"])
+	frappe.db.add_index("Stock Ledger Entry", ["item_code", "warehouse"], "item_warehouse")
```

Comment:
```suggestion
	frappe.db.add_index("Stock Ledger Entry", ["warehouse", "item_code"], "item_warehouse")
```

Might be beneficial. Needs some more testing (or better yet a generic perf test suite. 😅  )